### PR TITLE
[refactor](shuffle) Simplify hash partitioning strategy

### PR DESCRIPTION
**Pull request by Gabriel39**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Gabriel39** [comment]
run buildall

**Comment by Gabriel39** [comment]
run buildall

**Comment by Gabriel39** [comment]
run buildall

**Comment by Gabriel39** [comment]
run buildall

File: be/src/pipeline/exec/exchange_sink_operator.cpp [hunk 1]
```diff
@@ -211,6 +206,20 @@ Status ExchangeSinkLocalState::init(RuntimeState* state, LocalSinkStateInfo& inf
...
+        _partition_count = channels.size();
+        _partitioner.reset(
+                new vectorized::HashPartitioner(channels.size(), &_split_block_hash_compute_timer));
+        RETURN_IF_ERROR(_partitioner->init(p._texprs));
+        RETURN_IF_ERROR(_partitioner->prepare(state, p._row_desc));
+        RETURN_IF_ERROR(_partitioner->open(state));
+    } else if (p._part_type == TPartitionType::BUCKET_SHFFULE_HASH_PARTITIONED) {
+        _partition_count = channel_shared_ptrs.size();
+        _partitioner.reset(new vectorized::BucketHashPartitioner(channel_shared_ptrs.size()));
+        RETURN_IF_ERROR(_partitioner->init(p._texprs));
```
Comment (HappenLee) [inline be/src/pipeline/exec/exchange_sink_operator.cpp:219]:
```
 RETURN_IF_ERROR(_partitioner->init(p._texprs));
        RETURN_IF_ERROR(_partitioner->prepare(state, p._row_desc));
        RETURN_IF_ERROR(_partitioner->open(state));
```

better only write once.

File: be/src/vec/runtime/partitioner.cpp [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    if (rows > 0) {
+        {
+            SCOPED_CONSUME_MEM_TRACKER(mem_tracker);
+            RETURN_IF_ERROR(_get_partition_column_result(block, result));
+        }
+        SCOPED_TIMER(*_split_block_hash_compute_timer);
+        // result[j] means column index, i means rows index, here to calculate the xxhash value
+        for (int j = 0; j < result_size; ++j) {
+            // complex type most not implement get_data_at() method which column_const will call
+            unpack_if_const(block->get_by_position(result[j]).column)
```
Comment (HappenLee) [inline be/src/vec/runtime/partitioner.cpp:48]: abstract the function

**Comment by Gabriel39** [comment]
run buildall

**Review by zhangstar333 — APPROVED** [review]

**Review by Mryange — APPROVED** [review]

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**